File: workout-api/src/types/workout.ts
// src/types/workout.ts
export interface ParamsDictionary {
    id: string;
    workout_id?: string;
}

export interface WorkoutActivity {
    activity_id: number;
    sets: number;
    reps: number;
    weight?: number;
    duration?: number;
    distance?: number;
    date_time?: string;
}

export interface WorkoutRequestBody {
    name: string;
    description?: string;
    date: string;
    user_id: number;
    duration: number;
    activities?: WorkoutActivity[];
}

export interface ActivityInWorkoutBody {
    activity_id: number;
    sets: number;
    reps: number;
    weight?: number;
    duration?: number;
    distance?: number;
}

export interface Workout {
    id: number;
    name: string;
    description?: string;
    date: string;
    user_id: number;
    duration: number;
    activities?: WorkoutActivity[];
}

File: workout-api-spring/src/main/java/com/example/workoutapi/model/Workout.java
package com.example.workoutapi.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.util.Set;

@Data
@Entity
@Table(name = "workout")
public class Workout {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    private String description;
    private LocalDate date;
    private Double duration;
    
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToMany
    @JoinTable(
        name = "workout_activities",
        joinColumns = @JoinColumn(name = "workout_id"),
        inverseJoinColumns = @JoinColumn(name = "activity_id")
    )
    private Set<Activity> activities;
}